src/include/misc.h

```diff
@@ -274,10 +274,24 @@
  */
 #define WT_STREQ(s, cs) (sizeof(cs) == 2 ? (s)[0] == (cs)[0] && (s)[1] == '\0' : strcmp(s, cs) == 0)
 
-/* Check if a string matches a byte string of len bytes. */
-#define WT_STRING_MATCH(str, bytes, len)                                                        \
-    (((const char *)(str))[0] == ((const char *)(bytes))[0] && strncmp(str, bytes, len) == 0 && \
-      (str)[len] == '\0')
+/*
+ * Check if a string matches a byte string of len bytes. This macro expects both strings to be of
+ * non-zero length in order to be a match - this means that strings cannot be a match if len
+ * == 0.
+ *
+ * Use an inline function as that tends to be safer and we can check for NULL values.
```

Comment:
I got a little confused matching the text and the code, so I suggest rephrasing this.

First, "byte string" implies to me a string -- i.e., something null-terminated. But I think from looking at some uses of `WT_STRING_MATCH` it looks like it might not be a string. So maybe change `byte string` to `byte sequence`? Or even add a line saying that `bytes does not need to be null terminated`. 

Second, I would add the important bit that this returns false if len == 0 to the first line as well. Would the following be accurate:

`Check if a non-empty string matches a byte sequence of len bytes`. 

(The implication that an empty byte sequence, with len == 0, isn't a match for a non-empty string. If that seems too subtle you can certainly spell it out in the comment.)

Reply:
Thanks for the suggestions Keith, I'll change the comment to be more specific.